dist/jquery.flot.js

```diff
@@ -2459,13 +2459,33 @@ Licensed under the MIT license.
             return precision;
         }
 
-        function setEndpointTicks(axis) {
-            if (axis.options.showTickLabels === 'all' || axis.options.showTickLabels === 'endpoints') {
+        function setEndpointTicks(axis, series) {
+            if (isValidEndpointTick(axis, series)) {
                 axis.ticks.unshift(newTick(axis.min, null, axis, 'min'));
                 axis.ticks.push(newTick(axis.max, null, axis, 'max'));
             }
         }
 
+        function isValidEndpointTick(axis, series) {
+            if (axis.options.showTickLabels === 'endpoints') {
+                return true;
+            }
+            if (axis.options.showTickLabels === 'all') {
+                var axisSeries = series.filter(function(series) {
+                    return series.xaxis === axis || series.yaxis === axis;
+                });
```

Comment:
You can chain this filter() and an every() or some() which includes the logic from the next for statement.